File: montecarlo_pi/montecarlo.py
import math
import random


def inside(x, y):
    return x**2 + y**2 <= 1


insideCounter = 0
for i in range(0, 1000):
    if inside(random.random(), random.random()):
        insideCounter += 1

pi = insideCounter / 1000.0 * 4
print(pi)

File: montecarlo_pi/montecarlo.nim
import os, random

proc inside(): bool =
    let x = rand(1.0)
    let y = rand(1.0)

    return x*x + y*y <= 1

proc estimatePi(attempts: int): float =
    var hits = 0
    for i in 0..<attempts:
        if inside():
            inc hits

    var pi = hits.float / attempts.float * 4.0
    return pi


randomize()
let pi = estimatePi(2_000_000_000)
echo(pi)
